--- a/pyproject.toml
+++ b/pyproject.toml
@@ -3,10 +3,16 @@
 version = "0.1.0"
 requires-python = ">=3.13"
 dependencies = [
+    "astropy>=8.0.1",
+    "astroquery>=0.4.11",
     "matplotlib>=3.11.2",
+    "nbdime>=4.0.4",
     "numpy>=2.5.3",
     "pandas>=3.0.5",
+    "photutils>=3.0.0",
     "scikit-learn>=1.9.1",
+    "scipy>=1.18.1",
+    "specutils>=2.4.0",
     "torch>=2.14.0",
     "tqdm>=4.70.1",
 ]
@@ -52,6 +58,7 @@
 locator = "ml"

 [tool.protostar.tools]
+mypy = true

 [tool.protostar.fallback]
 ci = false
@@ -83,8 +90,22 @@
 [tool.protostar.metadata]

 [tool.protostar.bindings]
+
+# ---- Mypy ---- #
+
+[tool.mypy]
+mypy_path = "src"
+python_version = "3.13"
+pretty = true
+show_error_codes = true
+show_error_context = true
+warn_return_any = true
+warn_unused_configs = true
+check_untyped_defs = true
+explicit_package_bases = true
 [dependency-groups]
 dev = [
+    "mypy>=2.3.1",
     "ruff>=0.16.8",
 ]
 docs = [
